docs/students_and_interns.md

```diff
@@ -0,0 +1,31 @@
+
+## Outreachy and GSoC
+If you're thinking of applying to Outreachy and/or GSoC, please read this section!
+
+Outreachy and GSoC are paid, competitive programs where successful applicants join the Wiki Education Program team full time for several months to work on a specific project with support from mentors. We typically see a large number of folks show up in the weeks before the application deadline, ready to try their hand at fixing issues and deciding whether to apply for this project. We try to handle this process in a way that is as respectful as possible of the time, energy, and goals of everyone involved.
+
+### The application period
+The application period has two important roles: it lets us get to know you, the applicant; and it lets you get to know the project, its codebase, its development workflow, and its role in the Wikimedia ecosystem. If you're interested in applying, the first things you should do are to [set up a development environment](./setup.md) — and to join our Slack and say hi. After that, you should find a first issue to work on. (Please start with one, and wait until you've completed that to claim other issues.)
+
+For your early contributions, it may take more time to explain and review your work than it would take us to do the work ourselves. This is okay; these are learning opportunities for you, and that balance will shift over time if you continue contributing. 
+
+Advice for Outreachy & GSoC hopefuls:
+* The earlier you start contributing, the better chance you'll have. Learning the project takes time.
+* Please ask questions and seek help when you're stuck!
+* Before opening a pull request, rebase and fix any merge conflicts, and check that only the relevant files are being changed. There should be no unintended changes to the schema or dependency files, or other "git noise".  If you have a messy git history, squash your commits to clean it up. See [here](./git.md) for common git techniques and best practices.
+* When you're ready to submit code, read and follow the instructions for pull requests. To make code review easier on us, you should provide a detailed explanation of your changes, along with before-and-after screenshots or animations of what you worked on if there is a UI component to it. 
+* After you submit code, check the continuous integration build, which runs all the tests against your code. If there are failing tests, identify what went wrong and fix it if you can. (If you can't, ask for help.)
+* If you're trying to judge how to spend your time — whether to put in more work on this project, or start making contributions somewhere else to improve your chances — feel free to talk with us about it. We don't make any decisions before the final applications are complete, but we can give you a frank assessment of your chances.
```

Comment:
I think this section might need to split up a bit. Some of this is instructions (i.e. what to do first, pull requests instructions) while the other bit is advice. It might make the most amount of sense to keep the advice here and link to Pull Request instructions (which might go in the CONTRIBUTING.md file).